tests/unit/test_estimator.py

```diff
@@ -464,6 +477,50 @@ def test_estimator_initialization_with_sagemaker_config_injection(sagemaker_sess
     assert estimator.security_group_ids == expected_security_groups
     assert estimator.subnets == expected_subnets
     assert estimator.environment == expected_environment
+    assert estimator.debugger_hook_config == expected_debugger_hook_config
+
+
+def test_estimator_with_debugger_hook_config_provided_as_bool_from_direct_input(
```

Comment:
we can also add a UT where the config schema value is set to True in config.yaml, but no dict is provided by the user, the debug_hook_config params should have the required s3 path

Reply:
This is not a valid use-case since the documentation says to disable the config just provide `False` or either use a `dict` value to enable the `debugger_hook_config`.

Reply:
Check `_prepare_debugger_for_training()` function to understand more on the behavior of SDK for different values of `debugger_hook_config` field.

Reply:
No - I meant we set the debugger_hook_config value in config.yaml file to true. 
Then if the customer does not pass anything, the debug_hook_config should be populated with a default path correct? 

Similar to the UT in line 491, but the True value is set via the config.yaml file 

Or are you saying the user cannot set the debug_hook_config value to True in the config ? If they do, will it cause error ?

Reply:
Since the s3_output_path value is resolved in _prepare_debugger_for_training() function, it seems it doesn't take into account if the debugger_hook_config value when passed as True. SDK throws error if you set it to True.

Reply:
@akrishna1995 and I found out that this is actually a bug in SDK where is doesn't handled True as an input correctly. Fixed it.

Reply:
Thanks for the fix

offline discussion thread for reference: 
* We need to support boolean ( True and False ) in config.yaml
* For true value, we cannot throw error, we should have a default dict ( confirmed via table for debug hook config)
* If the current impl does not support True value, we should either change our schema to support only setting False or fix the impl - going with fix